engines/wasmtime/Dockerfile

```diff
@@ -1,10 +1,16 @@
 FROM rust:1.60
 ARG REPOSITORY=https://github.com/bytecodealliance/wasmtime/
 ARG REVISION=main
+ARG BUILD="cargo build -p wasmtime-bench-api"
+ARG FLAGS="--release"
 
 WORKDIR /usr/src
-RUN git clone ${REPOSITORY}
-WORKDIR /usr/src/wasmtime
-RUN git checkout ${REVISION} && git submodule update --init
-RUN cargo build -p wasmtime-bench-api --release
+# Clone a single commit (actually, a "revision"--tags and branches work as well).
+RUN git init
+RUN git remote add origin ${REPOSITORY}
+RUN git fetch --depth 1 origin ${REVISION}
```

Comment:
Cool. Haven't used this flag but I see it just does a shallow fetch and will reduce the size of the fetch (also take less time but that currently isn't an issue). Does this change though impact using the API to checkout a specific revision? I.e. .. 

'RUST_LOG=debug cargo run build-engine "wasmtime@${WASMTIME_BUILD}@https://github.com/bytecodealliance/wasmtime.git"'

Can you still checkout a specific revision with this change? Above should the variable REVISION really be renamed "branch"?

Reply:
Yes, `REVISION` can be either a branch, a commit, or a tag; at least that's what I've observed, more things might work there. The API shouldn't change at all.

Reply:
Wait... I should double-check that.

Reply:
Yup, it works for commits:

```
 1028  cd /tmp
 1029  mkdir wasmtime
 1030  cd wasmtime/
 1031  git init
 1032  git remote add origin git@github.com:bytecodealliance/wasmtime
 1033  git fetch --depth 1 origin e142f587a78613ffd30f52fd2cd11cd09e51af3b
 1034  git checkout FETCH_HEAD
 1035  git log
 ...
commit e142f587a78613ffd30f52fd2cd11cd09e51af3b (grafted, HEAD)
Author: Sam Parker <sam.parker@arm.com>
Date:   Thu Apr 14 20:16:56 2022 +0100

    [AArch64] Refactor ALUOp3 (#3950)
```